src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,55 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+interface Props {
+  filter: string;
+  query: string;
+  onFilterChange: (status: string) => void;
+  onSearchChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  onClearSearch: () => void;
+}
+
+export const TodoFilter: React.FC<Props> = ({ filter, query, onFilterChange, onSearchChange, onClearSearch }) => {
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            value={filter}
+            onChange={e => onFilterChange(e.target.value)}
+            data-cy="statusSelect"
+          >
```

Comment:
Consider adding a label for the select element to improve accessibility. This helps screen readers to associate the control with its purpose.